umbraco/Umbraco.CMS.Backoffice
# Feature: Adds new method to extensionRegistry for appendCondition & appendConditions

## Description

Adds two new methods to extensionRegistry to allow developers to append new conditions to an existing extension.

## Types of changes

## Motivation and context
Its not possible to update an extension such as Workspace Entity Action with a new condition. The current approach requires you to overwrite the extension. Which involves setting the same manifest meta properties and reimplementing an API to perform the same action as the core implementation of the button.

**Scenario:**
I want to show/hide the Save and Publish button based on a new condition and keep the existing functionality of the button without overhead of trying to overwrite the extension and keeping it in sync with the core implemetation.


## How to test?
* Read over the PR
* Read the associated tests
* Run the tests - ensure it passes

## Usage
```typescript
// Add a condition with a specific config to Section2
const workspaceCondition:WorkspaceAliasConditionConfig = {
	alias: 'Umb.Condition.WorkspaceAlias',
	match: 'Umb.Workspace.Document'
};
extensionRegistry.addCondition('Umb.Test.Section.2', workspaceCondition);
```

```typescript
const conditions:Array<UmbConditionConfigBase> = [
	{
		alias: 'Umb.Test.Condition.Valid'
	},
	{
		alias: 'Umb.Condition.WorkspaceAlias',
		match: 'Umb.Workspace.Document'
	} as WorkspaceAliasConditionConfig
]

extensionRegistry.addConditions('Umb.Test.Section.1', conditions);
```

## Checklist
